synapse/handlers/federation.py

```diff
@@ -269,6 +267,29 @@ async def _maybe_backfill_inner(
             sorted_backfill_points,
         )
 
+        # If we have no backfill points lower than the `current_depth` then
+        # either we can a) bail or b) still attempt to backfill. We opt to try
+        # backfilling anyway just in case we do get relevant events.
```

Comment:
This feels like a bit of an orthogonal change? I'm not sure whether we really want to do this

Reply:
Fair enough, filtering by `current_depth` in the query itself is an extra fix that we can be split out to another PR. And I can include the additional `limit` change I want to make in that future PR as well

For reference, the change is targeting not pulling out thousands of backfill points from the database every time we need to backfill. We only need to pull out a 20 relevant ones at most since we just use 5 of them.

And there is no functional change ([previous code](https://github.com/matrix-org/synapse/blob/f7c89c44c540d82c4e82a232e6ae1084b7b94e33/synapse/handlers/federation.py#L295-L320))

Reply:
Removed from this PR and split off to https://github.com/matrix-org/synapse/pull/13879 ⏩